pypeit/coadd3d.py

```diff
@@ -0,0 +1,1772 @@
+        # Return the scaling correction for this frame
+        return this_scalecorr, relScaleImg
+
+    def set_default_skysub(self):
+        """
+        Set the default mode to use for sky subtraction.
+        """
+        if self.cubepar['skysub_frame'] in [None, 'none', '', 'None']:
+            self.skysub_default = "none"
+            self.skyImgDef = np.array([0.0])  # Do not perform sky subtraction
+            self.skySclDef = np.array([0.0])  # Do not perform sky subtraction
+        elif self.cubepar['skysub_frame'].lower() == "image":
+            msgs.info("The sky model in the spec2d science frames will be used for sky subtraction" + msgs.newline() +
+                      "(unless specific skysub frames have been specified)")
+            self.skysub_default = "image"
+        else:
+            msgs.info("Loading default image for sky subtraction:" +
+                      msgs.newline() + self.cubepar['skysub_frame'])
+            try:
+                spec2DObj = spec2dobj.Spec2DObj.from_file(self.cubepar['skysub_frame'], self.detname)
+                skysub_exptime = fits.open(self.cubepar['skysub_frame'])[0].header['EXPTIME']
+                self.skysub_default = self.cubepar['skysub_frame']
+                self.skyImgDef = spec2DObj.sciimg / skysub_exptime  # Sky counts/second
+                # self.skyImgDef = spec2DObj.skymodel/skysub_exptime  # Sky counts/second
+                self.skySclDef = spec2DObj.scaleimg
+            except:
+                msgs.error("Could not load skysub image from spec2d file:" + msgs.newline() + self.cubepar['skysub_frame'])
+
+    def get_current_skysub(self, spec2DObj, exptime, opts_skysub=None):
+        """
+        Determine the sky frame that should be used to subtract from the science frame
+
+        Args:
+            spec2DObj (:class:`~pypeit.spec2dobj.Spec2DObj`_):
+                2D PypeIt spectra object.
+            exptime (:obj:`float`_):
+                The exposure time of the science frame (in seconds)
+            opts_skysub (:obj:`str`, optional):
+                A string that describes what mode should be used for the sky subtraction. The
+                allowed values are:
+                default - Use the default value, as defined in self.set_default_skysub()
+                image - Use the sky model derived from the science frame
+                none - Do not perform sky subtraction
+
+        Returns:
+            :obj:`str`_: A string that describes the sky subtration mode to be used (see opts_skysub description)
+            `numpy.ndarray`_: 2D image (same shape as science frame) containing the sky frame to be subtracted from the science frame
+            `numpy.ndarray`_: 2D image (same shape as science frame) containing the relative spectral scaling that has been applied to the returned sky frame
+        """
+        this_skysub = self.skysub_default
+        if self.skysub_default == "image":
```

Comment:
This control flow would be cleaner, if skysub_default were a parameter in a parset. Is there a way to do that, or is the decision making too complex?

Reply:
I don't think this should be a parset. In this case, `skysub_frame` is already in the parset, and sets the default usage. However, if a user wants to use a different skysub frame for different spec2d files, then we need to override the default behaviour.